app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala

```diff
@@ -55,10 +56,10 @@ object NSIUserInfo {
     * if successful.
     */
   def apply(userInfo: UserInfo): ValidatedNel[String, NSIUserInfo] = {
-    (forenameValidation(userInfo.forename) |@|
-      surnameValidation(userInfo.surname) |@|
+    (forenameValidation(transform(userInfo.forename)) |@|
+      surnameValidation(transform(userInfo.surname)) |@|
       dateValidation(userInfo.dateOfBirth) |@|
-      addressLineValidation(userInfo.address.lines) |@|
+      addressLineValidation(userInfo.address.lines.map(transform)) |@|
       postcodeValidation(userInfo.address.postcode) |@|
```

Comment:
I have now applied the transformer to the postcode and the country code. The nino regex is taking care of that.